src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IBoundDiscardOperation.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.IOperation)]
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [Fact]
+        public void DiscardExpression_AsAssignment()
```

Comment:
If discard can be used in a pattern or  a case clause, it would be good to test these scenarios as well.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/25841](https://github.com/dotnet/roslyn/issues/25841) to follow up on this, as it's a bigger problem then initially realized and beyond the scope of this PR.

---
In reply to: [178234455](https://github.com/dotnet/roslyn/pull/25820#discussion_r178234455) [](ancestors = 178234455)